File: src/types/company/company.ts
import {z} from "zod";
import {timestampSchema} from "@/types/timestamp";
import {ObjectId} from "mongodb";

export const companySchema = z.object({
  _id: z.custom<ObjectId>(),
  name: z.string().min(2, "En az 2 karakter gereklidir."),
  registrationAgreement: z.boolean(),
  userIds: z.array(z.string()),
  creatorUserId: z.string(),
  createdAt: timestampSchema,
  logoOriginalFileId: z.custom<ObjectId>().nullable().optional(),
  logoMiniFileId: z.custom<ObjectId>().nullable().optional(),
});

export type Company = z.infer<typeof companySchema>;

File: fisoft/src/main/kotlin/types/companies/Company.kt
package com.ilyasdemirkiran.types.companies

import kotlinx.serialization.Serializable
import org.bson.BsonType
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.codecs.pojo.annotations.BsonRepresentation

@Serializable
data class Company(
    @BsonId @BsonRepresentation(BsonType.OBJECT_ID) val _id: String? = null,
    val name: String? = "",
    val registrationAgreement: Boolean? = true,
    val userIds: List<String> = emptyList(),              // Firebase UIDs
    val creatorUserId: String? = null,                     // Firebase UID
    val createdAt: String? = "",
    @BsonRepresentation(BsonType.OBJECT_ID) val logoOriginalFileId: String? = null,
    @BsonRepresentation(BsonType.OBJECT_ID) val logoMiniFileId: String? = null
)

@Serializable
data class CompanyInvite(
    @BsonId @BsonRepresentation(BsonType.OBJECT_ID) val _id: String? = null,
    @BsonRepresentation(BsonType.OBJECT_ID) val companyId: String? = null,
    val inviterUserId: String? = null,                     // Firebase UID in MongoDB
    val creatorUserId: String? = null,                     // Firebase UID
    val invitedUserId: String? = null,                     // Firebase UID
    val invitedPhoneNumber: String? = null,
    val code: String? = null,
    val status: String? = "pending",
    val createdAt: String? = null,
    val updatedAt: String? = null
)

@Serializable
data class CompanyNote(
    @BsonId @BsonRepresentation(BsonType.OBJECT_ID) val _id: String? = null,
    @BsonRepresentation(BsonType.OBJECT_ID) val companyId: String? = null,
    val userId: String? = null,                            // Firebase UID
    val note: String? = "",
    val createdAt: String? = ""
)